_posts/2021/2021-06-17-apis-for-newbies.md

```diff
@@ -0,0 +1,106 @@
+This API provides you four different endpoints to get a JSON object of [1 random joke](https://official-joke-api.appspot.com/jokes/random){:target="_blank"} or an array of objects of [10 random jokes](https://official-joke-api.appspot.com/jokes/ten){:target="_blank"}. 
+The only thing left to do is to fetch the data and return the result.
+
+**This is one way how you can do that:**
+
+<iframe src="https://codesandbox.io/embed/fetch-data-from-api-b6vr3?fontsize=14&module=%2Fsrc%2FJokeList.tsx&theme=dark"
+    style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+    title="Fetch data from API"
+    allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+    sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+></iframe>
+
+## The One API (to rule them all)
+
+[The One API](https://the-one-api.dev/){:target="_blank"} is a massive database around the world of Lord of the Rings - I mean, how cool is that? 😱 
+You get data about the books by J. R. R. Tolkien and about the official movies by Peter Jackson as well as data about many characters and quotes. 
+The One API is very well explained and [documented](https://the-one-api.dev/documentation){:target="_blank"} which makes it easier as a newbie to kick-start. 
+You also have the possibility to add pagination, sorting, and filtering options to your API requests. 
+To get access to that awesome database, you need to register and then you get a bearer token. 
+Only the `/book` endpoint is available without authentication. 
+Furthermore, you have a limit to 100 requests every 10 minutes.
+
+**This is how you could fetch the data with your bearer token:**
+
+```js
+const TOKEN = "thisIsYourSecretToken"
+
+const fetchData = async () => {
+  const response = await fetch(
+    `https://the-one-api.dev/v2/character`,
+    {
+      headers: {
+        Authorization: `Bearer ${TOKEN}`
+      }
+    }
+  )
+  const result = await response.json();
+  return result;
+}
+```
+
+## The Edamam API
+
+[Edamam APIs](https://developer.edamam.com/){:target="_blank"} include Recipe Database, Nutrition Analysis, and Food Database APIs. 
+You need to register on the website to get access to all these informations and the calls are limited to 10 calls per minute. 
+I tried out the [Recipe Search API](https://developer.edamam.com/edamam-recipe-api){:target="_blank"}, which provides you massive data and a nice [documentation](https://developer.edamam.com/edamam-docs-recipe-api){:target="_blank"}. 
+You can play around with search and filters and the best thing is that you even get images. 
+By the way, I learned to know this API with a [Youtube Tutorial from Dev Ed](https://www.youtube.com/watch?v=U9T6YkEDkMo&ab_channel=DevEd) which I can highly recommend as well.
```

Comment:
"...,I got to know this APi because of a ..." (du hast es da kennengelernt)

or

"..., I learned this API by watching this..." (du hast es damit gelernt"